src/pyggdrasil/tree_inference/_simulate.py

```diff
@@ -581,3 +585,98 @@ def adjacency_to_root_dfs(
     root = list_tree_node[root_idx]
 
     return root
+
+
+def gen_sim_data(
+    params: dict,
```

Comment:
(Just repeating myself: I think Pydantic's `BaseModel` is the perfect candidate in use cases like this one.)